extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -44,57 +71,181 @@ export class PlatformService implements IPlatformService {
 		return process.platform;
 	}
 
-	copyFile(source: string, target: string): Promise<void> {
-		return fs.promises.copyFile(source, target);
+	outputChannelName(): string {
+		return this._outputChannel.name;
+	}
+
+	showOutputChannel(preserveFocus?: boolean): void {
+		this._outputChannel.show(preserveFocus);
 	}
 
-	fileExists(file: string): Promise<boolean> {
-		return fs.promises.access(file).then(() => {
+	osType(platform: string = this.platform()): OsType {
+		if (Object.values(OsType).includes(<OsType>platform)) {
+			return <OsType>platform;
+		} else {
+			return OsType.others;
+		}
+	}
+
+	async copyFile(source: string, target: string): Promise<void> {
+		return await fs.promises.copyFile(source, target);
+	}
+
+	async fileExists(file: string): Promise<boolean> {
+		try {
+			await fs.promises.access(file);
 			return true;
-		}).catch(error => {
+		} catch (error) {
 			if (error && error.code === 'ENOENT') {
 				return false;
 			}
 			throw error;
-		});
+		}
 	}
 
 	openFile(filePath: string): void {
 		vscode.commands.executeCommand('vscode.open', vscode.Uri.file(filePath));
 	}
 
-	showErrorMessage(message: string): void {
-		vscode.window.showErrorMessage(message);
+	showErrorMessage(error: Error | string): void {
+		vscode.window.showErrorMessage(getErrorMessage(error));
 	}
 
 	isNotebookNameUsed(title: string): boolean {
 		return (azdata.nb.notebookDocuments.findIndex(doc => doc.isUntitled && doc.fileName === title) > -1);
 	}
 
-	makeDirectory(path: string): Promise<void> {
-		return fs.promises.mkdir(path);
+	async makeDirectory(path: string): Promise<void> {
+		await fs.promises.mkdir(path);
 	}
 
-	readTextFile(filePath: string): Promise<string> {
-		return fs.promises.readFile(filePath, 'utf8');
+	async ensureDirectoryExists(directory: string = this.storagePath()): Promise<void> {
```

Comment:
Just realized you moved this one here, this function doesn't do what you would expect, it is assuming the parent folder exists. previously I had it as a private method in a context whether the parent folder is guaranteed to be there.  I'd recommend to document the behavior

2. this default parameter value is not appropriate, please remove the default value.

Reply:
I can change it to mkdir -p impmentation that will make the parent directories

Reply:
In my actual test I have found the root of storagepath to be created. Just resource-deployment subfolder was not created

Reply:
removed teh default value.